crates/ide_assists/src/handlers/generate_default_from_new.rs

```diff
@@ -0,0 +1,230 @@
+use crate::{
+    assist_context::{AssistContext, Assists},
+    AssistId,
+};
+use syntax::{
+    ast::{self, NameOwner},
+    AstNode, SyntaxKind, SyntaxNode, SyntaxText,
+};
+use test_utils::mark;
+
+// Assist: generate_default_from_new
+//
+// Generates default implementation from new method.
+//
+// ```
+// struct Example { _inner: () }
+//
+// impl Example {
+//     pub fn n$0ew() -> Self {
+//         Self { _inner: () }
+//     }
+// }
+// ```
+// ->
+// ```
+// struct Example { _inner: () }
+//
+// impl Example {
+//     pub fn new() -> Self {
+//         Self { _inner: () }
+//     }
+// }
+//
+// impl Default for Example {
+//     fn default() -> Self {
+//         Self::new()
+//     }
+// }
+// ```
+pub(crate) fn generate_default_from_new(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let fn_node: ast::Fn = ctx.find_node_at_offset()?;
+    let fn_name = fn_node.name()?.to_string();
+
+    if !fn_name.eq("new") {
+        mark::hit!(other_function_than_new);
+        return None;
+    }
+
+    if fn_node.param_list()?.params().count() != 0 {
+        mark::hit!(new_function_with_parameters);
+        return None;
+    }
+
+    let insert_after = scope_for_fn_insertion_node(&fn_node.syntax())?;
+    let impl_obj = ast::Impl::cast(insert_after)?;
+    let struct_name = impl_obj.self_ty()?.syntax().text();
+
+    let default_fn_syntax = default_fn_node_for_new(struct_name);
```

Comment:
We already have a helper function for generating `impl`s here https://github.com/rust-analyzer/rust-analyzer/blob/f682627da4be4777fa0c1527398ef4136cd929b1/crates/ide_assists/src/utils.rs#L406-L410 which also respects generics

This should also be called inside the closure in below as its only needed when the assist is executed